src/figureprinter/FigurePainter.java

```diff
@@ -2,98 +2,176 @@
+
+    public static void drawLeftUpTriangle(int length) {
         // TODO : complete the method
 
+        for (int i = 1; i<=length; i++) {
+            for (int j = 1; j <= i; j++) {
+                System.out.print("* ");
+            }
+            System.out.println();
+        }
     }
 
     /**
      * Draws the triangle by specified length number
      * for example when n = 5 , it must be such as
      * drawn below :
-     *             *
-     *           * *
-     *         * * *
-     *       * * * *
-     *     * * * * *
-     * @param length  specified length of triangle will be drawn
+     *           *
+     *         * *
+     *       * * *
+     *     * * * *
+     *   * * * * *
+     *
+     * @param length specified length of triangle will be drawn
      */
-    public static void drawRightUpTriangle(int length){
+    public static void drawRightUpTriangle(int length) {
         // TODO : complete the method
 
+
+        for (int i = 1; i <= length; i++) {
+            for (int j = length - i; j > 0; j--) {
+                System.out.print(" ");
+            }
+              for (int k = i; k > 0; k--) {
+                System.out.print("*");
+              }
+            System.out.println();
+        }
     }
+
     /**
      * Draws the triangle by specified length number
      * for example when n = 4 , it must be such as
      * drawn below :
+     * <p>
+     *   * * * *
+     *   * * *
+     *   * *
+     *   *
      *
-     *       * * * *
-     *       * * *
-     *       * *
-     *       *
-     *
-     * @param length  specified length of triangle will be drawn
+     * @param length specified length of triangle will be drawn
      */
-    public static void drawLeftBottomTriangle(int length){
+    public static void drawLeftBottomTriangle(int length) {
         // TODO : complete the method
+
+        for (int i = length; i >= 1; i--) {
+            for (int j = i; j >= 1; j--) {
+                System.out.print("* ");
+            }
+            System.out.print("\n");
+        }
     }
 
     /**
      * Draws the triangle by specified length number
      * for example when n = 4 , it must be such as
      * drawn below :
+     * <p>
+     *   * * * *
+     *     * * *
+     *       * *
+     *         *
      *
-     *       * * * *
-     *         * * *
-     *           * *
-     *             *
-     *
-     * @param length  specified length of triangle will be drawn
+     * @param length specified length of triangle will be drawn
      */
-    public static void drawRightBottomTriangle(int length){
+    public static void drawRightBottomTriangle(int length) {
         // TODO : complete the method
+
+        for (int i = length; i >= 1; i--) {
+            for (int j = length - i; j > 0; j--) {
+                System.out.print(" ");
+            }
+              for (int k = i; k > 0; k--) {
+                System.out.print("*");
+              }
+
+            System.out.println();
+        }
```

Comment:
for (int i = 0; i < length; i++) {

            for (int j = 0; j < i; j++) {
                System.out.print("  ");
            }
            for (int j = 0; j < length - i; j++) {
                System.out.print("* ");
            }
            System.out.println();
        }